File: web/src/api/daigou/proxyApi.js
import { respCode } from '@/utils/errorCode'
import service from '@/utils/request'

export const proxyRespCode = {
  ERROR: -1, //请求错误
  SUCCESS: 0, //请求成功
  FAIL: 1, //请求失败,一般指业务方面
  UNAUTHORIZED: 2, //鉴权失败
  EXPIRED: 3, //token失效(只在访问登录接口时使用)
  UNAUTHOPERATE: 4 //没有操作权限
}

const bathPath = 'daigou/proxyApi/'

// admin使用post请求代理访问server
async function proxyPost({ proxyURL, proxyData }) {
  const { code, msg, data } = await service({
    url: bathPath + 'proxyPost',
    method: 'post',
    data: {
      url: proxyURL,
      data: JSON.stringify(proxyData)
    }
  })
  if (code != respCode.SUCCESS) {
    this.$message.error(`proxyPost代理请求${proxyURL}错误,${msg}`)
    return
  }
  return data
}

export async function proxyGetShopCode(data) {
  return proxyPost({
    proxyURL: 'admin/getShopCode',
    proxyData: data
  })
}

export async function proxyUseShopCode(data) {
  return proxyPost({
    proxyURL: 'admin/useShopCode',
    proxyData: data
  })
}

File: server/app/api/dg_api/proxy_api.go
package dg_api

import (
	"daigou-admin/app/service/dg_service"
	"daigou-admin/global/request"
	"daigou-admin/global/response"
	"daigou-admin/utils"
	"daigou-admin/utils/respwrite"
	"daigou-admin/utils/zaplog"
	"fmt"

	"github.com/gin-gonic/gin"
	"github.com/gogf/gf/util/gvalid"
)

//代理访问

// @Tags 代理post访问
// @Summary admin代理访问server
// @Description admin使用post请求代理访问server
// @Produce  application/json
// @Accept  application/json
// @Param data body request.ProxyPostReq true "请求参数"
// @Success 200 {object} respwrite.JsonResponse{data=response.ProxyPostResp} "请求结果"
// @Router /daigou/proxyApi/proxyPost [POST]
func ProxyPost(c *gin.Context) {
	var (
		req  request.ProxyPostReq
		resp *response.ProxyPostResp
		err  error
	)

	//获取参数
	if err = c.ShouldBindJSON(&req); err != nil {
		zaplog.Errorf("api ProxyPost err:%w", err)
		respwrite.FailWithMessage(err.Error(), c)
		return
	}
	zaplog.Debugf("ProxyPost request:%s", utils.Dump(req))

	//检查参数
	if err = gvalid.CheckStruct(&req, nil); err != (*gvalid.Error)(nil) {
		if v, ok := err.(*gvalid.Error); ok {
			zaplog.Errorf("api ProxyPost err:%s", v.FirstString())
			respwrite.FailWithMessage(v.FirstString(), c)
		}
		return
	}

	//业务处理
	err, resp, msg := dg_service.ProxyPost(&req)
	if err != nil {
		zaplog.Errorf("api ProxyPost err:%s", err)
		respwrite.FailWithMessage(fmt.Sprintf("代理请求失败,%v", err), c)
		return
	}

	zaplog.Debugf("ProxyPost response:%s", utils.Dump(resp))
	respwrite.OkDetailed(resp, msg, c)
}
